client/src/actions/team.js

```diff
@@ -287,11 +278,11 @@ function teamDataAllocator(teamData) {
  */
 export const updateTeam = (teamId, teamData) => async (dispatch) => {
   try {
-    const { data } = await api.updateTeam(teamId, teamData);
-    const updatedData = teamDataAllocator(data);
+    const successMessage = await api.updateTeam(teamId, teamData)
```

Comment:
I see a potential design issue with requiring the api to determine the `successMessage`. Why place the burden on the API to determine what message to dispatch? Doesn't it make more sense for clients to examine the response from the API (which I believe we should go with the `updatedData` as before) and determine what message needs to be dispatched? This is a separation of concerns argument, and I think it'd be better for the client to determine what the message should be. 

I'd like more opinions on this though. @kdav108 @vgoh1

Reply:
I agree, the client should determine the message to show the user rather than depending on the backend because it's easier to check the HTTP code of the API call rather than message for maintainability reasons such as:
- not having to go through to the backend to deal with the success/error handling for the frontend if we have to make changes later,
- HTTP codes are a reliable way to tell whether the operation was successful or not which the frontend can use. And because we know what endpoint it's calling the success message can be handled by the frontend.